components/eamxx/src/share/field/field_impl.hpp

```diff
@@ -543,7 +543,7 @@ void Field::deep_copy_impl (const ST value) const {
     case 0:
       {
         auto v = get_view<ST,HD>();
-        v() = value;
+        Kokkos::deep_copy(v,value);
```

Comment:
I'm surprised there is an issue with this since `operator()` of views returns a reference, so `v() = value` should store in value in `v.data[0]`. Can you point to the line in the other PR that causes issues? Does the test you added here fail if you don't change this line?

Reply:
Oh sorry, I'm guessing the issue that you call this function on host trying to set the device view, you can't just access the element.